src/dft/backends/mklgpu/mklgpu_helpers.hpp

```diff
@@ -0,0 +1,142 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _ONEMKL_DFT_SRC_MKLGPU_HELPERS_HPP_
+#define _ONEMKL_DFT_SRC_MKLGPU_HELPERS_HPP_
+
+#include "oneapi/mkl/detail/exceptions.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+
+// MKLGPU header
+#include "oneapi/mkl/dfti.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+namespace mklgpu {
+namespace detail {
+
+/// Convert domain to equivalent backend native value.
+inline constexpr dft::domain to_mklgpu(dft::detail::domain dom) {
+    if (dom == dft::detail::domain::REAL) {
+        return dft::domain::REAL;
+    }
+    else {
+        return dft::domain::COMPLEX;
+    }
+}
+
+/// Convert precision to equivalent backend native value.
+inline constexpr dft::precision to_mklgpu(dft::detail::precision dom) {
+    if (dom == dft::detail::precision::SINGLE) {
+        return dft::precision::SINGLE;
+    }
+    else {
+        return dft::precision::DOUBLE;
+    }
+}
+
+/// Convert a config_param to equivalent backend native value.
+inline constexpr dft::config_param to_mklgpu(dft::detail::config_param param) {
+    using iparam = dft::detail::config_param;
+    using oparam = dft::config_param;
+    switch (param) {
+        case iparam::FORWARD_DOMAIN: return oparam::FORWARD_DOMAIN;
+        case iparam::DIMENSION: return oparam::DIMENSION;
+        case iparam::LENGTHS: return oparam::LENGTHS;
+        case iparam::PRECISION: return oparam::PRECISION;
+        case iparam::FORWARD_SCALE: return oparam::FORWARD_SCALE;
+        case iparam::NUMBER_OF_TRANSFORMS: return oparam::NUMBER_OF_TRANSFORMS;
+        case iparam::COMPLEX_STORAGE: return oparam::COMPLEX_STORAGE;
+        case iparam::REAL_STORAGE: return oparam::REAL_STORAGE;
+        case iparam::CONJUGATE_EVEN_STORAGE: return oparam::CONJUGATE_EVEN_STORAGE;
+        case iparam::INPUT_STRIDES: return oparam::INPUT_STRIDES;
+        case iparam::OUTPUT_STRIDES: return oparam::OUTPUT_STRIDES;
+        case iparam::FWD_DISTANCE: return oparam::FWD_DISTANCE;
+        case iparam::BWD_DISTANCE: return oparam::BWD_DISTANCE;
+        case iparam::WORKSPACE: return oparam::WORKSPACE;
+        case iparam::ORDERING: return oparam::ORDERING;
+        case iparam::TRANSPOSE: return oparam::TRANSPOSE;
+        case iparam::PACKED_FORMAT: return oparam::PACKED_FORMAT;
+        case iparam::COMMIT_STATUS: return oparam::COMMIT_STATUS;
+        default:
+            mkl::invalid_argument("dft", "MKLGPU descriptor set_value()", "Invalid config param.");
+            return static_cast<oparam>(0);
+    }
+}
+
+/** Convert a config_value to the backend's native value. Throw on invalid input.
+ * @Param The config param the value is for.
+ * @value THe config value to convert.
```

Comment:
```suggestion
 * @value The config value to convert.
```

Reply:
Done in [f5490a1](https://github.com/hjabird/oneMKL/pull/2/commits/f5490a1ec5b1e708820a6b711d8ca8200771e512).